controllers/front/ProductController.php

```diff
@@ -151,6 +151,9 @@ public function init()
                 header('HTTP/1.1 403 Forbidden');
                 header('Status: 403 Forbidden');
                 $this->errors[] = $this->trans('You do not have access to this product.', array(), 'Shop.Notifications.Error');
+                $this->setTemplate('errors/forbidden');
+
+                return;
```

Comment:
I understand that, but in `CategoryController` you have to return in order to avoid the next statements to be executed, while in `ProductController` there is a big `else {...}` preventing that.